externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala

```diff
@@ -73,6 +73,16 @@ object KDFRegistry {
     "string",
     "1.4.0")
 
+  val engine_url: KyuubiDefinedFunction = create(
+    "engine_url",
+    udf { () =>
+      Option(TaskContext.get()).map(_.getLocalProperty(KYUUBI_ENGINE_URL))
```

Comment:
Why not just call `KyuubiSparkUtil.engineUrl`

Reply:
Call `KyuubiSparkUtil.engineUrl` will throw an exception
```
Caused by: java.lang.IllegalStateException: No active or default Spark session found
        at org.apache.spark.sql.SparkSession$.$anonfun$active$2(SparkSession.scala:1073)
        at scala.Option.getOrElse(Option.scala:189)
        at org.apache.spark.sql.SparkSession$.$anonfun$active$1(SparkSession.scala:1073)
        at scala.Option.getOrElse(Option.scala:189)
        at org.apache.spark.sql.SparkSession$.active(SparkSession.scala:1072)
        at org.apache.kyuubi.engine.spark.KyuubiSparkUtil$.globalSparkContext(KyuubiSparkUtil.scala:38)
        at org.apache.kyuubi.engine.spark.KyuubiSparkUtil$.engineUrl(KyuubiSparkUtil.scala:57)
        at org.apache.kyuubi.engine.spark.udf.KDFRegistry$.$anonfun$engine_url$1(KDFRegistry.scala:79)
        ... 18 more
```

Reply:
`udf` is executed on the executor side.